res/controllers/Denon-HC1000S-scripts.js

```diff
@@ -0,0 +1,920 @@
+    DenonHC1000S.shift = ! DenonHC1000S.shift; // '!' inverts a boolean (true/false) value
+    print ("Shift is pressed " + DenonHC1000S.shift);
+    if ((DenonHC1000S.shiftSampleLeft && DenonHC1000S.shiftSampleRight && DenonHC1000S.shift) && (value === 64)) {
+        var samplersShow = engine.getValue("[Samplers]", "show_samplers");
+        print ("want to show Samplers in UI; samplersShow is " + samplersShow);
+        engine.setValue("[Samplers]", "show_samplers", !samplersShow);
+    }
+};
+
+DenonHC1000S.shiftSampleLeft = false;
+DenonHC1000S.shiftSampleLeftButton = function(channel, control, value, status, group) {
+    DenonHC1000S.shiftSampleLeft = ! DenonHC1000S.shiftSampleLeft; // '!' inverts a boolean (true/false) value
+    print ("shiftSampleLeft is pressed " + DenonHC1000S.shiftSampleLeft);
+    DenonHC1000S.LED_HotCue_and_Sampler(channel, control, value, status, group);
+    //DenonHC1000S.LED_Play_and_Cue(channel, control, value, status, group);
+    if ((DenonHC1000S.shiftSampleLeft && DenonHC1000S.shiftSampleRight && DenonHC1000S.shift) && (value === 64)) {
+        var samplersShow = engine.getValue("[Samplers]", "show_samplers");
+        print ("want to show Samplers in UI; samplersShow is " + samplersShow);
+        engine.setValue("[Samplers]", "show_samplers", !samplersShow);
+    }
+};
+
+DenonHC1000S.shiftSampleRight = false;
+DenonHC1000S.shiftSampleRightButton = function(channel, control, value, status, group) {
+    DenonHC1000S.shiftSampleRight = ! DenonHC1000S.shiftSampleRight; // '!' inverts a boolean (true/false) value
+    print ("shiftSampleRight is pressed " + DenonHC1000S.shiftSampleRight);
+    DenonHC1000S.LED_HotCue_and_Sampler(channel, control, value, status, group);
+    //DenonHC1000S.LED_Play_and_Cue(channel, control, value, status, group);
+    if ((DenonHC1000S.shiftSampleLeft && DenonHC1000S.shiftSampleRight && DenonHC1000S.shift) && (value === 64)) {
+        var samplersShow = engine.getValue("[Samplers]", "show_samplers");
+        print ("want to show Samplers in UI; samplersShow is " + samplersShow);
+        engine.setValue("[Samplers]", "show_samplers", !samplersShow);
+    }
+};
+
+
+///////
+
+DenonHC1000S.getValue = function(key) {
+    return engine.getValue(DenonHC1000S.group, key);
+};
+
+DenonHC1000S.setValue = function(key, value) {
+    engine.setValue(DenonHC1000S.group, key, value);
+};
+
+DenonHC1000S.samplesPerBeat = function(group) {
+    return 2 * engine.getValue(group, "track_samplerate") * 60 / engine.getValue(group, "file_bpm");
+};
+
+//==TrackSelect===================================================================
+
+DenonHC1000S.recvTrackSelectKnob = function(_channel, _control, value, _status) {
+    var knobDelta = DenonHC1000S.getKnobDelta(value);
+    if (DenonHC1000S.shift) {
+        var ctrlName = "ScrollVertical";
+    } else {
+        var ctrlName = "MoveVertical";
+    }
```

Comment:
194:13  error    'ctrlName' is already defined                                              no-redeclare

Scoping of `var` is extremly weird and unintutive, therefore I usually use `let` instead, but that's not available before Mixxx 2.4. I suggest to use a ternary operator here:

```suggestion
    var ctrlName = DenonHC1000S.shift ? "ScrollVertical" : "MoveVertical";
```